### ISSUE TITLE
InsertReturning cleanup

### ISSUE THREAD

**Issue by jamescheney** [links-lang/links#563]
While working on #558 I've noticed some dd things about InsertReturning (which allows us to retrieve the value of a field after insertion, which is especially useful for fields with default or automatically-populated values.)

We can't anymore write:
```
insert factorials values (f,i) [(f=32,i=43)] returning i;
```
because the parser rule that should handle this expects the list expression to be a `db_expression` instead of `exp`.  Easy to fix.

We can also write insert returning commands that insert zero or many rows, and get a runtime error.
```
links> insert factorials values (f,i) [] returning i;
*** Fatal error : InsertReturning: undefined for empty list of rows
links> insert factorials values (f,i) [(f=1,i=1),(f=2,i=2)] returning i;
*** Runtime error: Returned the wrong number of results executing insert into factorials("f","i") VALUES (1,1), (2,2) returning "i"
```
It might be better to either return a list of the inserted fields, or specialize the InsertReturning syntax/typing so that it is impossible to insert zero or multiple rows.  We don't appear to be making heavy (or any?) use of this and there are no tests for InsertReturning syntax or behavior, so this change seems low-impact.

- [x] fix syntax for InsertReturning
- [x] add test cases for InsertReturning

**Comment by jamescheney** [comment]
Edited to remove the "possibly change syntax to only allow inserting one thing".  Created separate issue for that.  The other things are fixed in #558.

**Comment by jamescheney** [comment]
Also, note that fixing the `db_expression` to `exp` problem caused shift-reduce conflicts.  The `exp` needs to be `typed_expression` instead, for future reference.

**Comment by jamescheney** [comment]
Closed by #564
